src/riak_ql_parser.yrl

```diff
@@ -1210,7 +1213,7 @@ assert_desc_key_field_type(DDL, ?SQL_PARAM{ name = [Name] }) ->
         _ ->
             return_error_flat(
                 "Elements in the local key marked descending (DESC) must be of "
-                "type sint64 or varchar, but was ~p.", [Type])
+                "an integer or binary type, but was ~p.", [Type])
```

Comment:
Same feeling. Specific error messages that directly relate to the error they are for will require more work from us, but could be read hundreds or thousands of times. General terms like integer type are not defined in the docs and doing so would be more work and would need to be read by the user before they see this error message to make sense.